VsGallery.WebService/Controllers/UploadController.cs

```diff
@@ -48,8 +49,16 @@ public IHttpActionResult AddOrUpdateVsix()
                 // Save the uploaded file to "UploadedFiles" folder
                 httpFile.SaveAs(fileSavePath);
 
+                var newFileName = httpFile.FileName.Split(new[] {".vsix"}, StringSplitOptions.None)[0] +
+                                  DateTime.UtcNow.ToString("yyyyMMdd").Replace("-", "");
+
+                var iteration = Directory.EnumerateFiles(_configuration.VsixStorageDirectory)
+                                    .Count(x => x.Split('\\').Last().StartsWith(newFileName,
+                                        StringComparison.OrdinalIgnoreCase)) + 1;
+
                 // Now move all the files uploaded to the main storage directory
-                File.Move(fileSavePath, Path.Combine(_configuration.VsixStorageDirectory, httpFile.FileName));
+                File.Move(fileSavePath,
```

Comment:
Ideally this change should only be applied if the user is actually uploading the second package with the same name, that way you preserve the original behavior of the system. Or even better, rename the older ones and keep the most recent version as the one with the original filename...

Reply:
Renaming of the old ones would probably be the better way to go about this I agree. I'll give it a change.